packages/diagram-ui/src/features/viewport/fitToScreenKeyboardListener.ts

```diff
@@ -0,0 +1,16 @@
+import { KeyListener, SModelElementImpl } from "sprotty";
+import { Action, CenterAction, FitToScreenAction } from "sprotty-protocol";
+import { matchesKeystroke } from "sprotty/lib/utils/keyboard.js";
+import { injectable } from "inversify";
+
+/**
+ * Copy of the sprotty built-in key listener for ctrl+shift+c/f, except we can now set custom options
+ */
+@injectable()
+export class CenterKeyboardListener extends KeyListener {
+    override keyDown(element: SModelElementImpl, event: KeyboardEvent): Action[] {
+        if (matchesKeystroke(event, "KeyC", "ctrlCmd", "shift")) return [CenterAction.create([], { retainZoom: true })];
```

Comment:
1. brackets pls
2. I would prefer if you check it like I did in the undo/redo thing, so that we don't need the sprotty internal (you ca see from the import) `matchesKeystroke`